grammar.js

```diff
@@ -103,6 +103,7 @@ module.exports = grammar({
     [$.binary_operation, $.unary_operation, $.field_expression],
     [$.binary_operation, $.field_expression],
     [$.list, $._pattern_atom],
+    [$._ident, $.lambda_expression],
```

Comment:
Why is this 'eeded now and was not needed before?

Reply:
Without this, I get a conflict
```
Unresolved conflict for symbol sequence:
  _function  '{'  identifier  •  '->'  …
Possible interpretations:
  1:  _function  '{'  (_ident  identifier)  •  '->'  …
  2:  _function  (lambda_expression  '{'  identifier  •  '->'  _expression  '}')
```
I don't exactly know where a `'{' identifier '->'` is allowed elsewhere than in lambda, but it doesn't know if it should consider the identifier as part of lambda expression or the `name` of the method expression

Reply:
Then it might be because of the braces expansions then.

There must be another place where this kind of conflict is handled.

Reply:
You would need double braces in a brace expansion if we use a lambda within it
```vim
let Function_name = {-> 'hello'}
echo Function_{{-> 'name'}()}()
```
also an identifier needs to immediately follow a `->` for it to be considered part of the method call. [`:h E274`](https://neovim.io/doc/user/eval.html#E274)

Reply:
I think I found where the conflict was coming from and why we didn't get it before.
By adding `method_expression`, we allowed the syntax
```
'{' _ident -> ... '}'
```
twice for an expression. One in `lamda_expression` and the other as a single curly_brace expression identifier
```
_ident (identifier (_curly_brace_name_expression '{' (method_expression _ident '->' _expression  '}' ))
```